src/compas/utilities/decorators.py

```diff
@@ -136,9 +136,129 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-# ==============================================================================
-# Main
-# ==============================================================================
+def _get_event_key(event_source, event_name):
+    return '/{}/{}'.format(event_source.__class__.__name__, event_name)
+
+
+def observable(observable_method=None, event_name=None, *args, **kwargs):
+    """Decorator to mark a property or method of a class as observable.
+
+    Observable methods/properties send notifications (events) that
+    other code can listen to and act when they are triggered. This
+    is useful to decouple classes where a class B depends on events
+    occurring on class A, but class A should not be in charge of
+    actively track and update class B on those events.
+
+    Notes
+    -----
+    This decorator should only be applied to methods and properties
+    of a class, not to stand-alone functions.
+
+    Examples
+    --------
+    >>> class Test(object):
+    ...   @observable(event_name='init')
```

Comment:
I trust myself to type 'init' correctly, but I do not trust myself to type 'updated_transformations' correctly.  I know this is what tests are for, but somethings are difficult to test (say, did the image of the robot  in blender update correctly when the rcf changed?). Is there a nice way of throwing an exception somewhere if the observable event names for a class don't match up with the observed event names or vice versa? or is that something that shouldn't be protected against?